### DISCUSSION TITLE
How can I set to prefer musl binaries and use gnu as fallback

### DISCUSSION THREAD

**Discussion by raivivek** [zdharma-continuum/zinit#527] [Q&A]
Hi! I'd like to setup my configuration such that it always downloads the musl binaries for a program through `gh-r` interface if available and only uses the `gnu` if not. Is there a way to configure this? This would only be for the Linux platform and in case of macOS, it should just use default behavior. I tried to define a variable and use that with `bpick` but it did not seem to work like how I've described here. Thanks!

**Comment by vladdoster** [comment]
@raivivek, 

Was your bpick `bpick'*musl*'`?

**Reply by raivivek** [reply]
Yes, that's the bpick I was using

**Comment by vladdoster** [comment]
It seems to be working when I tested in a Docker container. 

> [!NOTE]  
> By default, it uses `$OSTYPE` as the preference

Running a Ubuntu container:

```console
OSTYPE=linux-gnu MACHTYPE=aarch64 CPUTYPE=aarch64 hardware=aarch64 processor=unknown
```

> [!NOTE]  
> The last command in the screenshot is with no `bpick` ice (i.e., default behavior)

<img width="760" alt="Screenshot 2023-09-30 at 11 50 16 PM" src="https://github.com/zdharma-continuum/zinit/assets/10052309/f08724f7-6464-4623-a019-6a06475a1469">
